modules/gateway/conn.go

```diff
@@ -26,6 +26,7 @@ func (g *Gateway) dial(addr modules.NetAddress) (net.Conn, error) {
 	dialer := &net.Dialer{
 		Cancel:  g.threads.StopChan(),
 		Timeout: dialTimeout,
+		LocalAddr: config.Siad.HostAddr,
```

Comment:
close! But I think what we want is more like `LocalAddr: g.myAddr.Host()`. The part I'm not sure about is that (I believe) we need to specify the port as well. Maybe `:0` works, like it does with `net.Listen`?